contracts/finance/andromeda-weighted-distribution-splitter/src/contract.rs

```diff
@@ -0,0 +1,1145 @@
+            payload: encode_binary(&msg)?,
+        },
+    )?;
+
+    match msg {
+        ExecuteMsg::UpdateRecipients { recipients } => {
+            execute_update_recipients(deps, info, recipients)
+        }
+        ExecuteMsg::UpdateRecipientWeight { recipient } => {
+            execute_update_recipient_weight(deps, info, recipient)
+        }
+        ExecuteMsg::AddRecipient { recipient } => execute_add_recipient(deps, info, recipient),
+        ExecuteMsg::RemoveRecipient { recipient } => {
+            execute_remove_recipient(deps, info, recipient)
+        }
+        ExecuteMsg::UpdateLock { lock } => execute_update_lock(deps, info, lock),
+
+        ExecuteMsg::Send {} => execute_send(deps, info),
+        ExecuteMsg::AndrReceive(msg) => execute_andromeda(deps, env, info, msg),
+    }
+}
+
+pub fn execute_update_recipient_weight(
+    deps: DepsMut,
+    info: MessageInfo,
+    recipient: AddressWeight,
+) -> Result<Response, ContractError> {
+    // Only the contract's owner can update a recipient's weight
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // No need to send funds
+    require(
+        info.funds.is_empty(),
+        ContractError::FunctionDeclinesFunds {},
+    )?;
+    // Can't set weight to 0
+    require(
+        recipient.weight > Uint128::zero(),
+        ContractError::InvalidWeight {},
+    )?;
+
+    // Check if splitter is locked
+    let mut splitter = SPLITTER.load(deps.storage)?;
+
+    require(!splitter.locked, ContractError::ContractLocked {})?;
+
+    // Recipients are stored in a vector, we search for the desired recipient's index in the vector
+
+    let user_index = splitter
+        .recipients
+        .clone()
+        .into_iter()
+        .position(|x| x.recipient == recipient.recipient);
+
+    // If the index exists, change the element's weight.
+    // If the index doesn't exist, the recipient isn't on the list
+    require(user_index.is_some(), ContractError::UserNotFound {})?;
+
+    if let Some(i) = user_index {
+        splitter.recipients[i].weight = recipient.weight;
+        SPLITTER.save(deps.storage, &splitter)?;
+    };
+    Ok(Response::default().add_attribute("action", "updated_recipient_weight"))
+}
+
+pub fn execute_add_recipient(
+    deps: DepsMut,
+    info: MessageInfo,
+    recipient: AddressWeight,
+) -> Result<Response, ContractError> {
+    // Only the contract's owner can add a recipient
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // No need to send funds
+    require(
+        info.funds.is_empty(),
+        ContractError::FunctionDeclinesFunds {},
+    )?;
+    // Check if splitter is locked
+    let mut splitter = SPLITTER.load(deps.storage)?;
+
+    require(!splitter.locked, ContractError::ContractLocked {})?;
```

Comment:
This was one of the first things added to the splitter originally, the more I look at the idea of locking the splitter the more unsure how useful it really is. Do we think this is a useful feature?

Reply:
I can't think of a scenario where it's useful. This lock feature is like locking your iPhone's Face ID behind a pin code. Like you're the only person that can use the function you're locking. Requiring `is_owner_or_operator` is the actual lock.

Reply:
Actually, we can add a timer to the lock. Even the owner can't change it for 1 week let's say. That would serve as a guarantee for the recipients that they won't be removed, no new recipients will be added, and their weights will remain unchanged .